File: solutions/python/bob/1/bob.py
def response(hey_bob):
    hey_bob = hey_bob.strip()

    if not hey_bob:
        return "Fine. Be that way!"
    is_question = hey_bob.endswith("?")
    is_yelling = hey_bob.isupper()
    if is_question and is_yelling:
        return "Calm down, I know what I'm doing!"
    if is_question:
        return "Sure."
    if is_yelling:
        return "Whoa, chill out!"
    return "Whatever."

File: solutions/julia/bob/1/bob.jl
function bob(stimulus)
    statement = strip(stimulus)
    question = isquestion(statement)
    yelling = isyelling(statement)
    silent = isempty(statement)
    ( question && yelling ) ? "Calm down, I know what I'm doing!" : 
               ( question ) ? "Sure." :
                ( yelling ) ? "Whoa, chill out!" :
                 ( silent ) ? "Fine. Be that way!" :
                              "Whatever."
end

function isquestion(str)
    !isempty(str) && str[end] == '?'
end

function isyelling(str)
    any(isuppercase, str) && !any(islowercase,str)
end